smqtk_iqr/utils/web.py

```diff
@@ -51,25 +51,21 @@ def __init__(self, url):
             url = 'http://' + url
         self.url = url
 
-    def _compose(self, endpoint):
+    def _compose(self, endpoint: str) -> str:
         return '/'.join([self.url, endpoint])
 
-    def get(self, endpoint, **params):
+    def get(self, endpoint: str, **params: Optional[str]) -> requests.Response:
         # Make params None if its empty.
-        params = params and params or None
         return requests.get(self._compose(endpoint), params)
 
-    def post(self, endpoint, **params):
+    def post(self, endpoint: str, **params: Optional[str]) -> requests.Response:
         # Make params None if its empty.
-        params = params and params or None
         return requests.post(self._compose(endpoint), data=params)
 
-    def put(self, endpoint, **params):
+    def put(self, endpoint: str, **params: Optional[str]) -> requests.Response:
         # Make params None if its empty.
-        params = params and params or None
         return requests.put(self._compose(endpoint), data=params)
 
-    def delete(self, endpoint, **params):
+    def delete(self, endpoint: str, **params: Optional[str]) -> requests.Response:
```

Comment:
I'd like to confirm that these `Optional[str]` types for `params` are correct. I assumed that optional was implied with the usage of `**params`

Reply:
Not necessarily. Remember that `**params` equates to a `dict`. It is only assumed that the keys of that `dict` will be of `str` type. The value type is not assumed and that is what this type annotation is for. so the annotation here serves to annotated the possible values of the dict. We should annotate the transitive expectation of the `requests.*` methods that `params` is being passed in to.

Reply:
Diving deep down into requests, it seems that `params`, when in dictionary form like we have all four methods expect values to either be a string or an iterable of strings. If you want to trace, used pycharm to dive into `requests.models.PreparedRequest` and methods `prepare_url` and `prepare_body`.